Mathlib/CategoryTheory/MorphismProperty/Lifting.lean

```diff
@@ -0,0 +1,143 @@
+  ∀ ⦃X Y : C⦄ ⦃g : X ⟶ Y⦄ (_ : T g), HasLiftingProperty f g
+
+/-- The right lifting property (rlp) with respect to a class of morphisms. -/
+def rlp (T : MorphismProperty C) : MorphismProperty C := fun _ _ f ↦
+  ∀ ⦃X Y : C⦄ ⦃g : X ⟶ Y⦄ (_ : T g), HasLiftingProperty g f
+
+open RetractArrow in
+lemma liftOfRetractLeft {X Y Z W Z' W' : C} {f : X ⟶ Y} {g : Z ⟶ W} {g' : Z' ⟶ W'}
+    (h : RetractArrow g' g) (h' : HasLiftingProperty g f) : HasLiftingProperty g' f where
+  sq_hasLift := fun {u v} sq ↦
+    ⟨h.i.right ≫ h'.lift ⟨show (h.r.left ≫ u) ≫ f = g ≫ (h.r.right ≫ v) by simp [sq.w]⟩,
+      by rw [← Category.assoc, ← i_w, Category.assoc, CommSq.fac_left, ← Category.assoc,
+        retract_left, Category.id_comp],
+      by rw [Category.assoc, CommSq.fac_right, ← Category.assoc, retract_right, Category.id_comp]⟩
+
+open RetractArrow in
+lemma liftOfRetractRight {X Y Z W X' Y' : C} {f : X ⟶ Y} {f' : X' ⟶ Y'} {g : Z ⟶ W}
+    (h : RetractArrow f' f) (h' : HasLiftingProperty g f) : HasLiftingProperty g f' where
+  sq_hasLift := fun {u v} sq ↦
+    ⟨h'.lift ⟨show (u ≫ h.i.left) ≫ f = g ≫ (v ≫ h.i.right)
+      by rw [← Category.assoc, ← sq.w, Category.assoc, i_w, Category.assoc]⟩ ≫ h.r.left,
+      by rw [← Category.assoc, CommSq.fac_left, Category.assoc, retract_left, Category.comp_id],
+      by rw [Category.assoc, r_w, ← Category.assoc, CommSq.fac_right, Category.assoc, retract_right,
+        Category.comp_id]⟩
+
+lemma llp.IsStableUnderRetracts {T : MorphismProperty C} : T.llp.IsStableUnderRetracts :=
+  ⟨fun {_ _} _ _ _ _ h hg _ _ _ hf ↦ liftOfRetractLeft h (hg hf)⟩
+
+lemma rlp.IsStableUnderRetracts {T : MorphismProperty C} : T.rlp.IsStableUnderRetracts :=
+  ⟨fun {_ _} _ _ _ _ h hf _ _ _ hg ↦ liftOfRetractRight h (hf hg)⟩
+
+open IsPushout in
+instance llp.IsStableUnderCobaseChange {T : MorphismProperty C} :
+    T.llp.IsStableUnderCobaseChange where
+  of_isPushout {_ _ _ _ f s _ t} P hf _ _ g hg := {
+    sq_hasLift := fun {u v} sq ↦ by
+      have w : (s ≫ u) ≫ g = f ≫ (t ≫ v) := by
+        rw [← Category.assoc, ← P.toCommSq.w, Category.assoc, Category.assoc, sq.w]
+      exact ⟨P.desc u ((hf hg).lift ⟨w⟩) (by rw [CommSq.fac_left]), P.inl_desc ..,
+        P.hom_ext (by rw [inl_desc_assoc, sq.w]) (by rw [inr_desc_assoc, CommSq.fac_right])⟩}
+
+open IsPullback in
+instance rlp.IsStableUnderBaseChange {T : MorphismProperty C} :
+    T.rlp.IsStableUnderBaseChange where
+  of_isPullback {_ _ _ _ t f s _} P hf _ _ g hg := {
+    sq_hasLift := fun {u v} sq ↦ by
+      have w : (u ≫ s) ≫ f = g ≫ v ≫ t := by
+        rw [Category.assoc, P.toCommSq.w, ← Category.assoc, ← Category.assoc, sq.w]
+      exact ⟨P.lift ((hf hg).lift (.mk w)) v (by rw [CommSq.fac_right]),
+        P.hom_ext (by rw [Category.assoc, lift_fst, CommSq.fac_left])
+          (by rw [Category.assoc, lift_snd, sq.w]), P.lift_snd _ _ _⟩}
+
+instance llp.ContainsIdentities (T : MorphismProperty C) : T.llp.ContainsIdentities :=
```

Comment:
We can use the known results about `HasLiftingProperty`:
```suggestion
instance llp_isMultiplicative (T : MorphismProperty C) : T.llp.IsMultiplicative where
  id_mem X _ _ p hp := by infer_instance
  comp_mem i j hi hj _ _ p hp := by
    have := hi hp
    have := hj hp
    infer_instance
```

Reply:
Thanks, I hadn't looked very closely at the `HasLiftingProperty` API when I was writing this.